TileRarities.lua

```diff
@@ -0,0 +1,131 @@
+--[[
+    The data for the different chances and multiplier values can be found on Data:Tile_Chances
+]]--
+local p = {}
+
+function p.main(frame)
+	return p._main(frame.args)
+end
+
+function p._main(args)
+	local json = mw.loadJsonData('Data:Tile_Chances')
+    local ranks = json.ranks
+	local dom = mw.html.create()
```

Comment:
this is fine if you want to call it this, my preferred term here is `output` though, because then you know your code needs to end in `return output` (for me a common mistake is to forget to return anything).

in a lot of cases your life will be a bit easier if you follow my conventions because then youll have an easier time copying other code i wrote or reviewed since the variable names will be the same. up to you though!